samples/auth-user-sample/authUserFlow.js

```diff
@@ -0,0 +1,156 @@
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+'use strict';
+
+const {OAuth2Client} = require('google-auth-library');
+const readline = require('readline');
+
+function startRl() {
+  const rl = readline.createInterface({
+    input: process.stdin,
+    output: process.stdout,
+  });
+
+  rl.on('close', function() {
+    process.exit(0);
+  });
+
+  return rl;
+}
+
+// Download your OAuth2 configuration from the Google
+const keys = require('./oauth2.keys.json');
+
+/**
+ * Create a new OAuth2Client, and go through the OAuth2 content
+ * workflow. Return the full client to the callback.
+ */
+async function getRedirectUrl() {
+  const rl = main.startRl();
+  // Create an oAuth client to authorize the API call.  Secrets are kept in a `keys.json` file,
+  // which should be downloaded from the Google Developers Console.
+  const oAuth2Client = new OAuth2Client(
+    keys.installed.client_id,
+    keys.installed.client_secret,
+    keys.installed.redirect_uris[0]
+  );
+
+  // Generate the url that will be used for the consent dialog.
+  const authorizeUrl = oAuth2Client.generateAuthUrl({
+    access_type: 'offline',
+    scope: 'https://www.googleapis.com/auth/bigquery',
+    prompt: 'consent',
+  });
+
+  console.info(
+    `Please visit this URL to authorize this application: ${authorizeUrl}`
+  );
+  rl.question('Enter the authorization code: ', function(code) {
+    main.exchangeCode(code).then(function() {
+      rl.close();
+    });
+  });
+}
+
+// Exchange an authorization code for an access token
+async function exchangeCode(code) {
+  // Which should be downloaded from the Google Developers Console.
+  const oAuth2Client = new OAuth2Client(
+    keys.installed.client_id,
+    keys.installed.client_secret,
+    keys.installed.redirect_uris[0]
+  );
+
+  try {
+    const r = await oAuth2Client.getToken(code);
+    console.info(r.tokens);
+    return r.tokens;
+  } catch (err) {
+    console.log(err);
+  }
+}
+
+// [START bigquery_auth_user_flow]
+function authFlow(projectId = 'project_id') {
+  /**
+   * TODO(developer):
+   * Save Project ID as environment variable PROJECT_ID="project_id"
+   * Uncomment the following line before running the sample.
+   */
+  // const projectId = process.env.PROJECT_ID;
+
+  const credentials = {
+    type: 'authorized_user',
+    client_id: keys.installed.client_id,
+    client_secret: keys.installed.client_secret,
+    refresh_token: process.env.REFRESH_TOKEN,
+  };
+
+  return {
+    projectId,
+    credentials,
+  };
+}
+// [END bigquery_auth_user_flow]
+// [START bigquery_auth_user_query]
+async function query() {
+  const {BigQuery} = require('@google-cloud/bigquery');
+  const credentials = main.authFlow();
+  const bigquery = new BigQuery(credentials);
+  // Queries the U.S. given names dataset for the state of Texas.
+  const query = ` SELECT name, SUM(number) as total
```

Comment:
Probably ok to use the single quotes here, so you don't need to escape the ticks.

Reply:
I think it needs to be these ticks because it's consistent with the rest of the queries in the node samples.

Reply:
Generally you only use the backtick when you want to do string interpolation, like `hello ${name}`.  Otherwise, it's generally customary to use single quotes.

Reply:
I think this library uses the backticks specifically for the SQL queries. It doesn't work if you use single or double quotes instead.

Reply:
We've been taking this approach with our SQL samples, because it allows us to spread them
across multiple lines; would you be open to us sticking to this approach for consistency?

Reply:
It's fine 🤷‍♂